File: algorithms/1013. Pairs of Songs With Total Durations Divisible by 60/solution.cc
class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        vector<int> counter(60);
        int result = 0;
        for ( int t : time ) {
            result += counter[(60 - t % 60) % 60];
            counter[t % 60]++;
        }
        return result;
    }
};

File: algorithms/1013. Pairs of Songs With Total Durations Divisible by 60/solution.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        counter = [0] * 60
        result = 0
        for t in time:
            result += counter[(60 - t % 60) % 60]
            counter[t % 60] += 1
        return result
